extensions/BugModal/web/new_bug.js

```diff
@@ -0,0 +1,48 @@
+$(document).ready(function() {
+    bugzilla_ajax(
+            {
+                url: 'rest/bug_modal/products'
+            },
+            function(data) {
+                $('#product').empty()
+                $('#product').append($('<option>', { value: 'Select Product', text: 'Select Product' }));
+                // populate select menus
+                $.each(data.products, function(key, value) {
+                    $('#product').append($('<option>', { value: value.name, text: value.name }));
+                });
+            },
+            function() {}
```

Comment:
As of now, no, as this is just facilitating data loading in the drop-down. Maybe we can keep a small warning near to show that the data wasn't loaded. The previous error handling in bug_modal was to revert back to the non-edit mode, which we don't have here.